mu-plugins/central-hub/tests/phpunit/integration/meta-data/saveCustomFields.php

```diff
@@ -0,0 +1,195 @@
+	}
+
+	/**
+	 * Clean up the test environment after each test.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+		$config = [];
+		$_POST  = [];
+
+		// Clean up.
+		delete_post_meta( $this->post->ID, 'event-date' );
+		delete_post_meta( $this->post->ID, 'event-time' );
+		delete_post_meta( $this->post->ID, 'venue-name' );
+	}
+
+	/**
+	 * Test save_custom_fields() should delete post meta keys passed to empty $_POST array element when value equals
+	 * config delete state.
+	 */
+	public function test_should_delete_post_meta_keys_passed_to_empty_POST_array_element_when_value_equals_config_delete_state() {
+		// Custom fields config.
+		$config = [
+			'event-date' => [
+				'is_single'    => true,
+				'default'      => '',
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'event-time' => [
+				'is_single'    => true,
+				'default'      => '',
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'venue-name' => [
+				'is_single'    => true,
+				'default'      => '',
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+		];
+		// Add post meta to the database.
+		add_post_meta( $this->post->ID, 'event-date', '' );
+		add_post_meta( $this->post->ID, 'event-time', '' );
+		add_post_meta( $this->post->ID, 'venue-name', '' );
+		$_POST = [
+			'events' => [],
+		];
+
+		save_custom_fields( $config, 'events', $this->post->ID );
+
+		// Check that post meta keys are deleted when their value equals the config delete state.
+		$this->assertSame( [], $_POST['events'] );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'event-date', true ) );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'event-time', true ) );
+		$this->assertSame( '', get_post_meta( $this->post->ID, 'venue-name', true ) );
+	}
+
+	/**
+	 * Test save_custom_fields() should delete actual post meta values set in $_POST when they equal the config delete
+	 * state.
+	 */
+	public function test_should_delete_actual_post_meta_values_set_in_POST_when_they_equal_the_config_delete_state() {
+		// Custom fields config.
+		$config = [
+			'event-date' => [
+				'is_single'    => true,
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'event-time' => [
+				'is_single'    => true,
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+			'venue-name' => [
+				'is_single'    => true,
+				'delete_state' => '',
+				'sanitize'     => 'sanitize_text_field',
+			],
+		];
+		// Add post meta to the database.
+		add_post_meta( $this->post->ID, 'event-date', '' );
+		add_post_meta( $this->post->ID, 'event-time', '' );
+		add_post_meta( $this->post->ID, 'venue-name', '' );
+		// $_POST contains the actual value for each custom field.
+		$_POST = [
+			'events' => [
+				'event-date' => '',
+				'event-time' => '',
+				'venue-name' => '',
+			],
+		];
+
+		save_custom_fields( $config, 'events', $this->post->ID );
```

Comment:
@rgadon107 Let's double check that the post meta exists in the database _before_ running the save function.

```php
$this->assertSame( '2019-10-01', get_post_meta( $this->post->ID, 'event-date', true ) );
$this->assertSame( '10:05:37', get_post_meta( $this->post->ID, 'event-time', true ) );
$this->assertSame( 'Some venue name', get_post_meta( $this->post->ID, 'venue-name', true ) );
```

Note: I added some values. Make sure the values you store above are the ones you expect here.

Reply:
Done. Test passed.